app/abilities/person_readables.rb

```diff
@@ -21,37 +22,43 @@ def initialize(user, group = nil)
 
   private
 
-  def group_accessible_people
-    if read_permission_for_this_group?
-      can :index, Person,
-        group.people.only_public_data { |_| true }
-
-    elsif layer_and_below_read_in_above_layer?
-      can :index, Person,
-        group.people.only_public_data.visible_from_above(group) { |_| true }
-
-    elsif contact_data_visible?
-      can :index, Person,
-        group.people.only_public_data.contact_data_visible { |_| true }
-    end
-  end
+  attr_reader :roles_join
 
   def accessible_people
     if user.root?
       Person.only_public_data.then do |scope|
-        group ? scope.joins(roles: :group).distinct : scope
+        group ? scope.joins(roles_join).distinct : scope
       end
     else
       scope = Person.only_public_data.where(accessible_conditions.to_a).distinct
       if has_group_based_conditions?
         # Only add these joins when really necessary, because they are extremely expensive to
         # compute when there are a lot of people and roles
-        scope = scope.joins(roles: :group).where(groups: {deleted_at: nil})
+        scope = scope.joins(roles_join).where(groups: {deleted_at: nil})
       end
       scope
     end
   end
 
+  def group_accessible_people
+    if read_permission_for_this_group?
+      can :index, Person,
+        group_people.only_public_data { |_| true }
+
+    elsif layer_and_below_read_in_above_layer?
+      can :index, Person,
+        group_people.only_public_data.visible_from_above(group) { |_| true }
+
+    elsif contact_data_visible?
+      can :index, Person,
+        group_people.only_public_data.contact_data_visible { |_| true }
+    end
+  end
+
+  def group_people
+    Person.joins(roles_join).where(groups: {id: group.id})
```

Comment:
Soweit ich das sehe wird überall wo dieser Scope verwendet wird auch noch `.only_public_data` angehängt. Das könnte entsprechend gleich hier passieren. Bzw. könnte eine Methode `people_roles_scope` mit `Person.joins(roles_join).only_public_data` erstellt werden, welche sowohl in `accessible_people` wie auch in `group_accessible_people` verwendet wird.

Reply:
Hab ich extrahiert, ich glaub es hat aber noch potential, da hier den Fall group.nil? 1x im Konstruktor aber auch in der `people_roles_scope` methode behandeln.